src/ChargeClustering/ChargeClustering.jl

```diff
@@ -130,3 +145,78 @@ function group_points_by_distance(pts::AbstractVector{CartesianPoint{T}}, energi
     clustersidx, elem_ptr = _group_points_by_distance(pts, group_distance)
     VectorOfVectors(pts[clustersidx], elem_ptr), VectorOfVectors(energies[clustersidx], elem_ptr), VectorOfVectors(radius[clustersidx], elem_ptr)
 end
+
+function st_dbscan(pos::AbstractMatrix, time::AbstractVector;
+                   eps_space::Real, eps_time::Real,
+                   min_neighbors::Int = 1, min_cluster_size::Int = 1,
+                   leafsize::Int = 20)
+
+    N = size(pos, 1)
+    visited = falses(N)
+    labels = zeros(Int, N)
+    clusters = Vector{Clustering.DbscanCluster}()
+    cluster_id = 0
+    
+    # --- Convert position input to Float64 matrix if needed ---
+    if eltype(pos) <: CartesianPoint
+        pos = hcat([SVector(p.x, p.y, p.z) for p in pos]...)'
+    elseif eltype(pos) <: StaticVector
+        pos = hcat(pos...)'   # convert Vector of SVector{3,T} to Nx3 Matrix
+    elseif !(pos isa AbstractMatrix{<:AbstractFloat})
+        error("Unsupported position format: must be a matrix of floats or vector of points")
+    end

```

Comment:
Do we still need this or is this handled is `cluster_detector_hits` already.
Because you require your input to this method to already be an `AbstractMatrix`.